extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/SyncRedisAPIImpl.java

```diff
@@ -0,0 +1,1039 @@
+
+    @Override
+    public Response zinterstore(List<String> args) {
+        return await(redisAPI.zinterstore(args));
+    }
+
+    @Override
+    public Response zlexcount(String arg0, String arg1, String arg2) {
+        return await(redisAPI.zlexcount(arg0, arg1, arg2));
+    }
+
+    @Override
+    public Response zpopmax(List<String> args) {
+        return await(redisAPI.zpopmax(args));
+    }
+
+    @Override
+    public Response zpopmin(List<String> args) {
+        return await(redisAPI.zpopmin(args));
+    }
+
+    @Override
+    public Response zrange(List<String> args) {
+        return await(redisAPI.zrange(args));
+    }
+
+    @Override
+    public Response zrangebylex(List<String> args) {
+        return await(redisAPI.zrangebylex(args));
+    }
+
+    @Override
+    public Response zrangebyscore(List<String> args) {
+        return await(redisAPI.zrangebyscore(args));
+    }
+
+    @Override
+    public Response zrank(String arg0, String arg1) {
+        return await(redisAPI.zrank(arg0, arg1));
+    }
+
+    @Override
+    public Response zrem(List<String> args) {
+        return await(redisAPI.zrem(args));
+    }
+
+    @Override
+    public Response zremrangebylex(String arg0, String arg1, String arg2) {
+        return await(redisAPI.zremrangebylex(arg0, arg1, arg2));
+    }
+
+    @Override
+    public Response zremrangebyrank(String arg0, String arg1, String arg2) {
+        return await(redisAPI.zremrangebyrank(arg0, arg1, arg2));
+    }
+
+    @Override
+    public Response zremrangebyscore(String arg0, String arg1, String arg2) {
+        return await(redisAPI.zremrangebyscore(arg0, arg1, arg2));
+    }
+
+    @Override
+    public Response zrevrange(List<String> args) {
+        return await(redisAPI.zrevrange(args));
+    }
+
+    @Override
+    public Response zrevrangebylex(List<String> args) {
+        return await(redisAPI.zrevrangebylex(args));
+    }
+
+    @Override
+    public Response zrevrangebyscore(List<String> args) {
+        return await(redisAPI.zrevrangebyscore(args));
+    }
+
+    @Override
+    public Response zrevrank(String arg0, String arg1) {
+        return await(redisAPI.zrevrank(arg0, arg1));
+    }
+
+    @Override
+    public Response zscan(List<String> args) {
+        return await(redisAPI.zscan(args));
+    }
+
+    @Override
+    public Response zscore(String arg0, String arg1) {
+        return await(redisAPI.zscore(arg0, arg1));
+    }
+
+    @Override
+    public Response zunionstore(List<String> args) {
+        return await(redisAPI.zunionstore(args));
+    }
+
+    private Response await(Uni<io.vertx.mutiny.redis.client.Response> mutinyResponse) {
+        try {
+            return mutinyResponse.subscribeAsCompletionStage()
+                    .toCompletableFuture()
+                    .get(timeout, SECONDS)
+                    .getDelegate();
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            throw new RuntimeException(e);
```

Comment:
you need to reset the interrupt flag if you get an InterruptedException.

In case of an ExecutionException, the cause must be wrapped not the exception.

Reply:
I guess it is not needed anymore with this https://github.com/quarkusio/quarkus/pull/9370#discussion_r443355401 suggestion.